src/transactions/ParallelApplyUtils.cpp

```diff
@@ -472,34 +472,10 @@ GlobalParallelApplyLedgerState::getSnapshotLedgerSeq() const
     return mLiveSnapshot->getLedgerSeq();
 }
 
-std::optional<LedgerEntry>
-GlobalParallelApplyLedgerState::getLiveEntryOpt(LedgerKey const& key) const
+ParallelApplyEntryMap const&
+GlobalParallelApplyLedgerState::getGlobalEntryMap() const
 {
-    auto it0 = mGlobalEntryMap.find(key);
-    if (it0 != mGlobalEntryMap.end())
-    {
-        return it0->second.mLedgerEntry;
-    }
-    // Invariant check: If an entry was restored from the live state, then it's
-    // possible that the global entry map does not have that key (because live
-    // restores only update the ttl), but if the entry was restored from the hot
-    // archive, both the ttl entry and the entry itself are updated. So if the
-    // key is missing from the global entry map, it could not have been
-    // previously restored from the hot archive.
-    releaseAssertOrThrow(!mGlobalRestoredEntries.entryWasRestoredFromMap(
-        key, mGlobalRestoredEntries.hotArchive));
-
-    // Check InMemorySorobanState cache for soroban types
-    std::shared_ptr<LedgerEntry const> res;
-    if (InMemorySorobanState::isInMemoryType(key))
-    {
-        res = mInMemorySorobanState.get(key);
-    }
-    else
-    {
-        res = mLiveSnapshot->load(key);
-    }
-    return res ? std::make_optional(*res) : std::nullopt;
+    return mGlobalEntryMap;
```

Comment:
great opportunity for a [TSA PreCheck](https://www.tsa.gov/travel/frequently-asked-questions/what-difference-between-global-entry-tsa-precheckr-and-other) reference here